validator/sawtooth_validator/execution/executor.py

```diff
@@ -77,47 +102,188 @@ def run(self):
+                default_value="[]")
+            # After reading the transaction families required in configuration
+            # try to json.loads them into a python object
+            # If there is a misconfiguration, proceed as if there is no
+            # configuration.
+            try:
+                transaction_families = json.loads(transaction_families)
+                required_transaction_processors = [
+                    processor_iterator.ProcessorType(
+                        d.get('family'),
+                        d.get('version'),
+                        d.get('encoding')) for d in transaction_families]
+            except ValueError:
+                LOGGER.warning("sawtooth.validator.transaction_families "
+                               "misconfigured. Expecting a json array, found"
+                               " %s", transaction_families)
+                required_transaction_processors = []
+
+            config_processor = processor_iterator.ProcessorType(
+                name='sawtooth_config',
+                version='1.0',
+                encoding='application/protobuf')
+
+            # If there are required transaction processors, sawtooth_config
+            # should be one of them, so the TransactionExecutor doesn't
+            # automatically fail config transactions.
+            if len(required_transaction_processors) > 0 and config_processor \
+                    not in required_transaction_processors:
+                required_transaction_processors += [config_processor]
+
+            if processor_type not in required_transaction_processors:
+                # If there aren't any configured required transaction
+                # processors, wait for all transactions. This is useful
+                # for startup with the assumption being that the
+                # required transaction processors would be configured soon
+                # after startup.
+                if len(required_transaction_processors) == 0:
+                    # There are no required txn processors
+                    LOGGER.debug("there are no required transaction "
+                                 "processors")
+                    self._execute_or_wait_for_processor_type(
+                        processor_type=processor_type,
+                        content=content)
+                else:
+                    # The txn processor type is not in the required
+                    # transaction processors so
+                    # failing transaction right away
+                    LOGGER.debug("failing transaction of type (name=%s,"
+                                 "version=%s,encoding=%s) since it isn't"
+                                 " required in the configuration",
+                                 processor_type.name,
+                                 processor_type.version,
+                                 processor_type.encoding)
+                    self._context_manager.delete_context(
```

Comment:
As an optimization, the context creation could have been skipped in the case that we aren't intending to execute  the transaction.